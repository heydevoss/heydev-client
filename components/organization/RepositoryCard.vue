<template>
  <div class="card">
    <el-card shadow="hover">
      <div slot="header" class="card--header">
        <span class="repository--name">{{ repository.name }}</span>
      </div>
      <div class="repository--counters">
        <i class="el-icon-star-on">{{ repository.totalStars }}</i>
        <i class="el-icon-warning">{{ repository.totalOpenIssues }}</i>
        <img width="15px" height="15px" src="/fork.png" />
        {{ repository.totalForks }}
      </div>
    </el-card>
  </div>
</template>

<script>
export default {
  name: 'Repository',
  props: {
    repository: Object
  }
}
</script>

<style lang="scss">
.card {
  min-width: 200px;
  min-height: 200px;
}
</style>
