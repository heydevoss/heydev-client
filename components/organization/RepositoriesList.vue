<template>
  <div>
    <ApolloQuery
      :query="require('@/apollo/queries/getRepositories.gql')"
      :variables="{ quantity }"
    >
      <template v-slot="{ result: { loading, error, data } }">
        <div v-if="loading" v-loading="loading" class="loading apollo" />

        <div v-else-if="error" class="error apollo">An error occured</div>

        <!-- Result -->
        <div v-else-if="data" class="repositories">
          <div
            v-for="repository in data.organization.repositories"
            :key="repository.id"
            class="repository"
          >
            <RepositoryCard :repository="repository" />
          </div>
        </div>
        <div v-else v-loading="true" class="loading" />
      </template>
    </ApolloQuery>
    <el-button class="button--show-more" @click="showMoreRepositories"
      >Show more</el-button
    >
  </div>
</template>
<script>
import RepositoryCard from '@/components/organization/RepositoryCard'
export default {
  name: 'RepositoriesList',
  components: {
    RepositoryCard
  },
  data() {
    return {
      quantity: 6
    }
  },
  methods: {
    showMoreRepositories() {
      this.quantity += 3
    }
  }
}
</script>
<style lang="scss" scoped>
.repositories {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
.repository {
  margin: 3px 3px;
  width: 300px;
}

.loading {
  display: flex;
  justify-content: center;
}
</style>
