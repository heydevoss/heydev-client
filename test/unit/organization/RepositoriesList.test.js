import { mount } from '@vue/test-utils'
import RepositoriesList from '@/components/organization/RepositoriesList.vue'
import RepositoryCard from '@/components/organization/RepositoryCard.vue'

describe('<RepositoriesList/>', () => {
  const wrapper = mount(RepositoriesList, {
    propsData: {
      quantity: 1
    },
    data: () => ({
      getRepositories: {
        organization: {
          repositories: [
            {
              name: 'MyTestRepository',
              totalStars: 20,
              totalForks: 10,
              totalOpenIssues: 3
            }
          ]
        }
      }
    }),
    stubs: {
      'el-card': true,
      'el-button': '<button></button>',
      'repository-card': RepositoryCard,
      ApolloQuery: true
    }
  })

  it('should be a Vue instance and renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })
})
