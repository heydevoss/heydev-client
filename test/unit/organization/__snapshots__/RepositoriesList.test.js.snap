// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoriesList/> should be a Vue instance and renders correctly 1`] = `
<div
  quantity="1"
>
  <apolloquery-stub
    query="[object Object]"
    variables="[object Object]"
  />
   
  <button
    class="button--show-more"
  />
</div>
`;
